src/client.js

```diff
@@ -163,6 +164,11 @@ function MatrixClient(opts) {
 
         this.olmVersion = Crypto.getOlmVersion();
     }
+
+    // Set up a sync accumulator if we can persist room data
+    if (this.store instanceof IndexedDBStore) {
+        this._syncAccumulator = this.store.getSyncAccumulator();
+    }
```

Comment:
Surely if we're going all OO then go the whole hog and just implement a `getSyncAccumulator` on the other stores that returns `null`, rather than `instanceof`ing.